### ISSUE TITLE
Input Box Carrot Rendering Issues

### ISSUE THREAD

**Issue by SneakyTactician** [SneakyTactician/MagicalLife#15]
The carrot of input boxes displays incorrectly, or off centered. It bugs me.

**Comment by SneakyTactician** [comment]
Here's where the location to display the carrot is calculated at: 

https://github.com/SneakyTactician/MagicalLife/blob/master/MagicalLifeGUIWindows/Rendering/GUI/GUIRenderer.cs 
CalculateCarrotBounds(MonoInputBox textbox, GUIContainer container)

**Comment by kanyewestknight** [comment]
{X:0 Y:0}
{X:5.5 Y:9}

first is where the carrot starts off, 2nd is after typing. Changing the following values did not change anything with how the carrot works:

Vector2 pos = new Vector2(textbox.DrawingBounds.Center.X, textbox.DrawingBounds.Center.Y);
#pragma warning disable RCS1096 // Use bitwise operation instead of calling 'HasFlag'.
            if (textbox.TextAlignment.HasFlag(Alignment.Left))
            {
                origin.X += (textbox.DrawingBounds.Width * 2) - (size.X / 2);
            }
            if (textbox.TextAlignment.HasFlag(Alignment.Right))
            {
                origin.X -= (textbox.DrawingBounds.Width / 2) - (size.X / 2);
            }
            if (textbox.TextAlignment.HasFlag(Alignment.Top))
            {
                origin.Y += (textbox.DrawingBounds.Height / 2) - (size.Y / 2);
            }
            if (textbox.TextAlignment.HasFlag(Alignment.Bottom))
            {
                origin.Y -= (textbox.DrawingBounds.Height / 2) - (size.Y / 2);
            }

The only location I could get the carrot to move initially was:
            int XPos = (int)Math.Round(origin.X + textbox.DrawingBounds.X + textbox.Font.MeasureString(TextBeforeCarrot).X) + container.DrawingBounds.X;
            int YPos = (int)Math.Round(origin.Y + textbox.DrawingBounds.Y) + container.DrawingBounds.Y;

but it would stay that way even after typing (so changing the X would keep it X distance away the entire time typing). Continuing to work on this.

**Comment by SneakyTactician** [comment]
Yeah, it's a tough one. I tried to fix it a month ago, and got nowhere.
